### PR TITLE
Tidy up rgbkb/mun

### PR DESCRIPTION
**Pull request by zvecr**
## Description

* Remove via from default keymap
* Avoid duplication of matrix code while keeping portscan
* Remove unused HAL enables

@XScorpion2 If you could have a look over the changes, that would be awesome.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rgbkb/mun/mcuconf.h
Comment (zvecr) [inline keyboards/rgbkb/mun/mcuconf.h:21]: I think this can also be removed, as TIM4 isnt currently being used?

**Review by XScorpion2 — APPROVED** [review]
looks fine

File: keyboards/rgbkb/mun/mcuconf.h
Comment (XScorpion2) [reply keyboards/rgbkb/mun/mcuconf.h:21]: I thought I needed this for the WS2812 driver

Comment (drashna) [reply keyboards/rgbkb/mun/mcuconf.h:21]: No driver is specified for WS2812, from what  can see. So it defaults to bitbang, which doesn't use a timer.

Comment (zvecr) [reply keyboards/rgbkb/mun/mcuconf.h:21]: Has `#define WS2812_PWM_DRIVER PWMD3`, so I assume TIM4 is left over from some other tinkering.

**Review by tzarc — APPROVED** [review]

File: keyboards/rgbkb/mun/mcuconf.h
Comment (XScorpion2) [reply keyboards/rgbkb/mun/mcuconf.h:21]:
```
#define WS2812_PWM_DRIVER PWMD3
#define WS2812_PWM_CHANNEL 2
#define WS2812_PWM_PAL_MODE 2
#define WS2812_DMA_STREAM STM32_DMA1_STREAM3
#define WS2812_DMA_CHANNEL 3
```

From my understanding, that requires one of the timers to operate. I thought it was TIM4

Comment (XScorpion2) [reply keyboards/rgbkb/mun/mcuconf.h:21]: Either way I'll double check everything this weekend and can resolve any issues.

**Review by drashna — APPROVED** [review]

File: keyboards/rgbkb/mun/mcuconf.h
Comment (drashna) [reply keyboards/rgbkb/mun/mcuconf.h:21]: I think that would be timer 3? But i could be wrong. 

But unless `WS2812_DRIVER = pwm` is set, then it won't use the pwm driver, and will default to the bitbang one. 

Did you mean to use the PWM driver for the ws2812?

Comment (XScorpion2) [reply keyboards/rgbkb/mun/mcuconf.h:21]: @drashna it's using pwm now, please see original pr here:
https://github.com/qmk/qmk_firmware/pull/13239/files#diff-0ea3f1d2d976dc67361be11cdc194440a7b54056abb950b257a00d750e07698eR25

**Comment by XScorpion2** [comment]
Please do not merge this until I can test the code changes locally this weekend, especially with the up coming develop > master merge.

File: keyboards/rgbkb/mun/mcuconf.h
Comment (drashna) [reply keyboards/rgbkb/mun/mcuconf.h:21]: ... i don't know how  messed the pwm bit

Comment (zvecr) [reply keyboards/rgbkb/mun/mcuconf.h]: You would get compile issues as PWMD4 was being used but not defined, if removing it was an issue. I guess my question was more why would it still need to be enabled when nothing is using it, rather than isolating that TIM3 is being used for ws2812.

Removing it after having had a 2nd look is definitely the right thing in regards to the code that currently exists on develop.

**Comment by zvecr** [comment]
> especially with the up coming develop > master merge.

It is for that reason that this should be merged, as its target is to tidy up the keyboard **before** it is merged to master and slightly more "locked in". We still accept but fixes to develop, so anything that might have broken can still be fixed.

**Pull request merged**
